docs/notes/knowledge_base.rst

```diff
@@ -196,6 +196,34 @@ Step type      Discount | Value used          | Value          Note
 ============== ======== ===================== ================ ===================================================
 
 
+DataTransformers
+-----------
+
+A DataTransformer takes in data from rollout or replay, does some processing and
+returns the modified data.
+
+It is a useful abstraction to organize all kinds of data processing. For example,
+``ObservationNormalizer`` normalizes input data to be zero mean and one std.
+
+However, it is important to note that ``HindsightExperienceTransformer``,
+``FrameStacker`` or any data transformer that need to access the replay buffer
+for additional data need to be before all other data transformers.
```

Comment:
"that need to access the replay buffer for additional data" this sentence is still too vague to understand.

Reply:
improved a bit and added more example.